### ISSUE TITLE
Very Slow Training

### ISSUE THREAD

**Issue by cryptedp** [kenshohara/3D-ResNets-PyTorch#26]
I am training Resnet with depth 34 on the kinetics dataset, however the training procedure is not improving anything. How long does it take till the model starts improving ? I have attached a screenshot; currently I am at epoch 34 but the loss is still 5.99 and is not decreasing, and accuracy is very volatile

![selection_121](https://user-images.githubusercontent.com/5487736/35319116-b81ed10c-00de-11e8-810e-19bbebe1b308.png)

**Comment by kenshohara** [comment]
Training and validation loss in our training of ResNet-18 are shown in Figure 4 (d) in my paper.
Your results are strange...
Could you tell me your options in the training?

**Comment by cryptedp** [comment]
My options are : python main.py --root_path /data/sammer/ --video_path video_directory/train/jpg --annotation_path kinetics.json --result_path results --dataset kinetics --model resnet --model_depth 34 --n_classes 400 --batch_size 64 --n_threads 4 --checkpoint 5 (so basically as mentioned in the readme)

**Comment by kenshohara** [comment]
Could you print all contents of opts.json?
Did you use our utils scripts to prepare datasets?

**Comment by cryptedp** [comment]
What I did: I downloaded the Kinetics dataset using the crawler (the train dataset) and put it into video_directory/train, converted them to jpg and then i called the n_frames script(n_frames is always under the corresponding jpg directory). In the end I created a json and saved it.

My opts.json is like this:
{"root_path": "/data/sammer/", "video_path": "/data/sammer/video_directory/train/jpg", "annotation_path": "/data/sammer/kinetics.json", "result_path": "/data/sammer/results", "dataset": "kinetics", "n_classes": 400, "n_finetune_classes": 400, "sample_size": 112, "sample_duration": 16, "initial_scale": 1.0, "n_scales": 5, "scale_step": 0.84089641525, "train_crop": "corner", "learning_rate": 0.1, "momentum": 0.9, "dampening": 0.9, "weight_decay": 0.001, "mean_dataset": "activitynet", "no_mean_norm": false, "std_norm": false, "nesterov": false, "optimizer": "sgd", "lr_patience": 10, "batch_size": 64, "n_epochs": 200, "begin_epoch": 1, "n_val_samples": 3, "resume_path": "", "pretrain_path": "", "ft_begin_index": 0, "no_train": false, "no_val": false, "test": false, "test_subset": "val", "scale_in_test": 1.0, "crop_position_in_test": "c", "no_softmax_in_test": false, "no_cuda": false, "n_threads": 4, "checkpoint": 5, "no_hflip": false, "norm_value": 1, "model": "standard", "model_depth": 34, "resnet_shortcut": "B", "wide_resnet_k": 2, "resnext_cardinality": 32, "manual_seed": 1, "scales": [1.0, 0.84089641525, 0.7071067811803005, 0.5946035574934808, 0.4999999999911653], "arch": "standard-34", "mean": [114.7748, 107.7354, 99.475], "std": [38.7568578, 37.88248729, 40.02898126]}

**Comment by kenshohara** [comment]
```--model``` of this opts.json is ```standard```.
Is this the same as ```resnet```?

Did you converted avi files to jpg files using my ```video_jpg_kinetics.py```,
or used your own script?
If you use your own script, resolutions of jpg files may be different.

**Comment by cryptedp** [comment]
Standard is the normal C3D model; I trained on it first, aborted it and then trained on resnet. Should I delete opts json and retrain it? Otherwise, yes it is your scripts

**Comment by kenshohara** [comment]
Previous opts.json is not a problem.
But I cannot find your problem...
Could you share log when you execute the script?

**Comment by cryptedp** [comment]
Hm i restarted it after having deleted opts json and it seems to be better now
![image](https://user-images.githubusercontent.com/5487736/35430329-142a9f2c-0278-11e8-9ce1-e41694f639a7.png)

**Comment by kenshohara** [comment]
Really? I cannot understand...
But I'm glad to hear that!

**Comment by iuhiyuh** [comment]
@cryptedp 
Has the problem been solved now?I met the same problem.

**Comment by suravijayjilla** [comment]
Hi,

How to generate opt JSON file.
I don't know how to generate that file, please guide me.

**Comment by guilhermesurek** [comment]
@suravijayjilla just create an empty json file and name it as 'opts.json' inside of results folder. When you run, the code overwrite the file.
